pkg/controllers/terminating/v1alpha1/controller.go

```diff
@@ -63,22 +65,68 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 	}
 }
 
-// Reconcile executes a reallocation control loop for the resource
+// Reconcile executes a termination control loop for the resource
+// Requeue a reconcile.Request if:
+//   1) Cordon, Drain, or Terminate fails
+//   2) Drain succeeds, but node has non-daemonset pods
 func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
-	// 1. Cordon terminable nodes
-	if err := c.terminator.cordonNodes(ctx); err != nil {
-		return reconcile.Result{}, fmt.Errorf("cordoning terminable nodes, %w", err)
+	node := object.(*v1.Node)
+	// 0. Don't do work on a non-terminable node
+	if phase, ok := node.Labels[provisioning.ProvisionerPhaseLabel]; !ok || phase == provisioning.ProvisionerUnderutilizedPhase {
+		return reconcile.Result{}, nil
 	}
 
-	// 2. Drain and delete nodes
-	if err := c.terminator.terminateNodes(ctx); err != nil {
-		return reconcile.Result{}, fmt.Errorf("terminating nodes, %w", err)
+	// 1. Cordon terminable nodes
+	if err := c.terminator.cordon(ctx, node); err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("cordoning terminable nodes, %w", err)
+	}
+	// 2. Drain node
+	drained, err := c.terminator.drain(ctx, node)
+	if err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("draining nodes, %w", err)
+	}
+	// 3. Terminate and Delete node
+	if drained {
+		if err := c.terminator.terminate(ctx, node); err != nil {
+			return reconcile.Result{Requeue: true}, fmt.Errorf("terminating nodes, %w", err)
+		}
 	}
-	return reconcile.Result{}, nil
+	// 4. Requeue if we still need to drain
+	return reconcile.Result{Requeue: !drained}, nil
 }
 
-func (c *Controller) Watches(context.Context) (source.Source, handler.EventHandler, builder.WatchesOption) {
+func (c *Controller) Watches() (source.Source, handler.EventHandler, builder.WatchesOption) {
 	return &source.Kind{Type: &v1.Node{}},
 		&handler.EnqueueRequestForObject{},
-		builder.WithPredicates(predicate.NewPredicateFuncs(func(object client.Object) bool { return false }))
+		builder.WithPredicates(
+			predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					return false
+				},
+				// Deleting nodes directly is not supported.
+				// Our termination design will add a webhook to validate delete requests, and then send an update call.
```

Comment:
You can probably drop both of these comments. Consider inlining all of these funcs as well
```
CreateFunc: func(e event.CreateEvent) bool { return false }, // no-op
DeleteFunc: func(e event.DeleteEvent) bool { return false }, // no-op
...
```